app/questionnaire/routing/when_rule_evaluator.py

```diff
@@ -36,12 +37,30 @@ def __post_init__(self) -> None:
             routing_path_block_ids=self.routing_path_block_ids,
             use_default_answer=True,
         )
+        operations = Operations()
+        self.operation_mapping: dict[str, Callable] = {
+            Operator.NOT: operations.evaluate_not,
+            Operator.AND: operations.evaluate_and,
+            Operator.OR: operations.evaluate_or,
+            Operator.EQUAL: operations.evaluate_equal,
+            Operator.NOT_EQUAL: operations.evaluate_not_equal,
+            Operator.GREATER_THAN: operations.evaluate_greater_than,
+            Operator.LESS_THAN: operations.evaluate_less_than,
+            Operator.GREATER_THAN_OR_EQUAL: operations.evaluate_greater_than_or_equal,
+            Operator.LESS_THAN_OR_EQUAL: operations.evaluate_less_than_or_equal,
+            Operator.IN: operations.evaluate_in,
+            Operator.ALL_IN: operations.evaluate_all_in,
+            Operator.ANY_IN: operations.evaluate_any_in,
+            Operator.COUNT: operations.evaluate_count,
+            Operator.DATE: operations.resolve_date_from_string,
+        }
 
     def _evaluate(self, rule: dict[str, Sequence]) -> Union[bool, Optional[date]]:
-        operator = Operator(next(iter(rule)))
-        operands = rule[operator.name]
+        next_rule = next(iter(rule))
+        operator = Operator(next_rule, self.operation_mapping[next_rule])
+        operands = rule[next_rule]
 
-        if not isinstance(operands, Sequence):
+        if not isinstance(rule[next_rule], Sequence):
```

Comment:
I think you forgot about using previously assigned `operands` here?

Reply:
Oh yeah, I changed it, then realised it was used elsewhere, but must of missed changing it back, thanks